Arrangement_on_surface_2/doc/Arrangement_on_surface_2/CGAL/Arr_caching_polyline_traits_2.h

```diff
@@ -0,0 +1,166 @@
+ * handles polylines as well. Use this alternative when you need to construct
+ * many polylines and every polyline consists of many points. Typically
+ * (depending on the substituted Kernel) the geometric objects that compose the
+ * polylines (e.g., points) are reference counted, which implies that a copy of
+ * a single point amounts to copying a small piece of data (a handle). However,
+ * when a polyline is constructed, split, or subdivided into \f$x\f$-monotone
+ * pieces, a large number of points might be copied, which yields with a costly
+ * operation. This traits class template uses an internal type to represent
+ * polylines, which in turn uses an internal type to represent lines and points
+ * that compose polylines. Polylines in the internal representation share the
+ * geometric data; thus, only one copy of the geometric objects that compose
+ * several polylines reside in memory.  Only a shallow copy is performed to
+ * carry out each one of the operations above.
+ *
+ * CAUTION: If you construct polylines using this traits class template and
+ * then insert the polylines into an arrangement, for instance, you must retain
+ * the polyline in memory as long as the arrangement is in memory.
+ *
+ * \tparam Kernel a type that represents a geometric kernel.  The number type
+ * used by the substituted kernel should support exact rational arithmetic (that
+ * is, the number type should support the arithmetic operations \f$ +\f$, \f$
+ * -\f$, \f$ \times\f$ and \f$ \div\f$ carried out without loss of precision),
+ * in order to avoid robustness problems, although other inexact number types
+ * could be used at the user's own risk.
+ *
+ * \tparam Range a type that represents a valid range of points.
+ *
+ * A polyline that comprises \f$n > 0\f$ segments has \f$ n+1 \f$ points, and
+ * they are represented as objects of type `Kernel::Point_2`. Since the
+ * notion of a \a vertex is reserved to 0-dimensional elements of an
+ * arrangement, we use, in this context, the notion of \a points in order to
+ * refer to the vertices of a polyline. For example, an arrangement induced by a
+ * single non-self intersecting polyline has exactly two vertices regardless of
+ * the number of points.
+ *
+ * \cgalModels `ArrangementTraits_2`
+ * \cgalModels `ArrangementDirectionalXMonotoneTraits_2`
+ * \cgalModels `ArrangementConstructXMonotoneCurveTraits_2`
+ * \cgalModels `ArrangementApproximateTraits_2`
+ *
+ * \sa `Arr_polyline_traits_2<SubcurveTraits_2>`
+ */
+
+template <typename Kernel, typename Range>
+class Arr_caching_polyline_traits_2 :
+    public Arr_polycurve_traits_2<SegmentTraits_2> {
+public:
+
+  /// \name Types
+  /// @{
+  /*!
+   */
+  typedef typename Kernel::Point_2                                Point_2;
+
+  /*!
+   */
+  typedef internal::X_monotone_caching_polyline_2<Kernel, Range>
+    X_monotone_curve_2;
+
+  /*!
+   */
+  typedef internal::X_monotone_caching_polyline_2<Kernel, Range>  Curve_2;
+  typedef
```

Comment:
In case you do want to keep the types documented, it's `internal::Caching_polyline_2` in that case (otherwise, `unspecified_type` too).

Also there's an isolated `typedef` here, probably a typo.